packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -543,6 +541,37 @@ function updateTextContent(vnode: VText | VComment, renderer: RendererAPI) {
     }
 }
 
+function insertFragmentOrNode(
+    vnode: VNode,
+    parent: Node,
+    anchor: Node | null,
+    renderer: RendererAPI
+) {
+    if (process.env.NODE_ENV !== 'production') {
+        unlockDomMutation();
+    }
+
+    if (isVFragment(vnode)) {
+        const children = vnode.children;
+        const nodesToInsert = [];
+        for (let i = 0; i < children.length; i += 1) {
+            const child = children[i];
+            if (!isNull(child)) {
+                nodesToInsert.push(child.elm!);
+            }
+        }
+        for (let i = 0; i < nodesToInsert.length; i += 1) {
+            renderer.insert(nodesToInsert[i], parent, anchor);
+        }
```

Comment:
These two `for` loops should be collapsible into one, to avoid creating the temporary `nodesToInsert` array.

Reply:
Yeah, I had them separated because the `nextSibling()` would be a different value once I inserted the node, but I think I can read the next sibling before inserting in the same loop.